backend/src/main/java/atlantbh/restaurants/models/filters/RestaurantFilterBuilder.java

```diff
@@ -39,26 +42,30 @@ public Criteria addConditions(Criteria rootCriteria, boolean isCountCriteria) {
             rootCriteria.createAlias("cousine", "cous");
             rootCriteria.createAlias("location", "loc");
             rootCriteria.add(criterion);
-        }
-        if (StringUtils.isNotBlank(name)) {
-            rootCriteria.add(Restrictions.ilike("name", name, MatchMode.ANYWHERE));
-        }
-        if (StringUtils.isNotBlank(category)) {
-            rootCriteria.createAlias("category", "cat");
-            rootCriteria.add(Restrictions.ilike("cat.name", category, MatchMode.ANYWHERE));
-        }
-        if (priceRange != null && priceRange > 0) {
-            rootCriteria.add(Restrictions.eq("priceRange", priceRange));
-        }
-        if (StringUtils.isNotBlank(location)) {
-            rootCriteria.createAlias("location", "loc");
-            Criterion criterion = Restrictions.or(Restrictions.ilike("loc.city", location, MatchMode.ANYWHERE),
-                    Restrictions.ilike("loc.country", location, MatchMode.ANYWHERE));
-            rootCriteria.add(criterion);
-        }
-        if (StringUtils.isNotBlank(cousine)) {
-            rootCriteria.createAlias("cousine", "c");
-            rootCriteria.add(Restrictions.ilike("c.name", cousine, MatchMode.ANYWHERE));
+        } else { // otherwise creating aliases would throw exception
+            if (StringUtils.isNotBlank(name)) {
+                rootCriteria.add(Restrictions.ilike("name", name, MatchMode.ANYWHERE));
+            }
+            if (StringUtils.isNotBlank(category)) {
+                rootCriteria.createAlias("category", "cat");
+                rootCriteria.add(Restrictions.ilike("cat.name", category));
+            }
+            if (priceRange != null && priceRange > 0 && priceRange <= 4) {
+                rootCriteria.add(Restrictions.le("priceRange", priceRange));
+            }
+            if (StringUtils.isNotBlank(location)) {
+                rootCriteria.createAlias("location", "loc");
+                Criterion criterion = Restrictions.or(Restrictions.ilike("loc.city", location, MatchMode.ANYWHERE),
+                        Restrictions.ilike("loc.country", location, MatchMode.ANYWHERE));
+                rootCriteria.add(criterion);
+            }
+            if (cousines != null && cousines.size() > 0) {
```

Comment:
CollectionUtils.isNotEmpty(cousines)

Reply:
Fixed in 0cd4debab433260ab6a8267aa0b72b31db581edb